edit_course/views.py

```diff
@@ -122,6 +124,22 @@ def renumber_recursion(self, parent, n=1):
             n += 1
         return n
 
+    def number_learning_objects_consecutively(self) -> None:
```

Comment:
This function gets called only when the form is updated through A+ UI, right? Is it a problem that consecutive_order is not having correct values until the user does submits the form? For reference, the old 'order' field is set also in managers.py, when new objects are created. Should consecutive_order be set there, too?

Reply:
This is a good question, since as you mentioned, if a new object is created, it is given the next order number, and the order numbers can also be manually edited for each learning object in the Edit Course view.
Probably most often new objects are created by editing the course rst files and importing them (in which case the consecutive_order is set in edit_course/operations/configure). Therefore this implementation is somewhat more of a backup allowing a teacher to renumber the learning objects consecutively while renumbering the learning objects. However, it's true that a teacher might not renumber the learning objects using the button if they just add a new learning object. However, when the consecutive_order is updated, it should be updated for all learning objects. I'm not sure how easy it would be to update all of them through managers.py (although it might be less bad to have some duplicates than have some missing the field entirely).

So good question, and I'm not sure.